### PR TITLE
RD-2586 use spot fleet

### PR DESCRIPTION
**Pull request by yuvlar**
After a long time:
First CI pipeline to use a cloudify blueprint that utilizes spot-fleet feature.
Updated BP to use spot fleet with pre-created roles, profiles and infra resources.

updated jenkinsfile with latest changes.

After this is quiet for a while, implementation will occur on other pipelines as well.

### CODE REVIEW

**Review by bartoszkosciug — COMMENTED** [review]
looks okay generally please just see my comment regarding naming of the stage step

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -336,15 +334,17 @@ EOT
     }
     stage('Terminate py27 instnace'){ // we made github to wait on a job with this exact name, typo included
```
Comment (bartoszkosciug) [inline jenkins/Jenkinsfile:335]: Shouldn't we change the py27 to py36?

Comment (yuvlar) [reply jenkins/Jenkinsfile:335]: yes indeed.
Change incoming in a minute

Comment (yuvlar) [reply jenkins/Jenkinsfile:335]: FYI, in such cases we also need to update the branch protection to match the stage name.

**Review by bartoszkosciug — DISMISSED** [review]
looks very good!

**Review by bartoszkosciug — APPROVED** [review]

**Pull request merged**